pymongo/monitoring.py

```diff
@@ -570,12 +570,14 @@ def __init__(
         connection_id: _Address,
         operation_id: Optional[int],
         service_id: Optional[ObjectId] = None,
+        database_name: Optional[str] = None,
```

Comment:
Why is database_name optional?

Reply:
I followed the conventions above `service_id`. I also assumed `Command(Succeeded|Failed)Event`s could be instantiated in some other codebase and fail because it's now a required argument. I figured keeping it optional doesn't introduce any backward breaking changes.

Reply:
Oh good point. I suppose one alternative would be to make it `database_name: str = "",`. What do you think about using `""` to mean no database was given (which could only happen if the user instantiated these classes on their own)? That would avoid the None/assert issue.

Reply:
Sounds good to me.